specification/subscription/resource-manager/Microsoft.Subscription/preview/2019-10-01-preview/subscriptions.json

```diff
@@ -392,6 +392,145 @@
           }
         ]
       }
+    },
+    "/providers/Microsoft.Subscription/subscriptions/{subscriptionId}": {
+      "put": {
+        "description": "The operation to create a new subscription or update the display name of an existing subscription.",
+        "operationId": "Subscription_PutSubscription",
+        "x-ms-examples": {
+          "createSubscription": {
```

Comment:
Can this API be used for an update? What is the behavior if a PUT is submitted for an already-existing subscription resource?

Reply:
We will be supporting update of the display name via this PUT - if same PUT is called today, we return the Subscription object (similar to the one in GET Subscription).

Reply:
Thanks, so please clarify:
1. If customer calls PUT without changing anything, you return a 200 OK with the existing resource (same as the GET body)?
2. If customer calls PUT with just the display name changed, you update the display name, and return a 200 OK with the new resource (including the display name change)?
3. If customer calls PUT with any other property added, updated, or removed, you return a 4XX response?

Reply:
1. yes, we return same response.
2. this is yet to implement, but from Subscription RP / Provisioning services perspective we update in our DB and can return 200, but ARM takes some time to reflect the name change across.
3. yes, we return 4XX.